bmc/redfish.go

```diff
@@ -182,14 +182,15 @@ func (r *RedfishBMC) GetSystemInfo(systemUUID string) (SystemInfo, error) {
 	}
 
 	return SystemInfo{
-		SystemUUID:   system.UUID,
-		Manufacturer: system.Manufacturer,
-		Model:        system.Model,
-		Status:       system.Status,
-		PowerState:   system.PowerState,
-		SerialNumber: system.SerialNumber,
-		SKU:          system.SKU,
-		IndicatorLED: string(system.IndicatorLED),
+		SystemUUID:             system.UUID,
+		Manufacturer:           system.Manufacturer,
+		Model:                  system.Model,
+		Status:                 system.Status,
+		PowerState:             system.PowerState,
+		SerialNumber:           system.SerialNumber,
+		SKU:                    system.SKU,
+		IndicatorLED:           string(system.IndicatorLED),
+		TotalSystemMemoryBytes: int64(system.MemorySummary.TotalSystemMemoryGiB) * 1024 * 1024 * 1024,
```

Comment:
To avoid the multiplication duding assignment we could switch to a `resource.Quanity` type as described above.

Reply:
We could do something like that:
```
	memoryString := fmt.Sprintf("%dGi", memoryGiB)
	quantity, err := resource.ParseQuantity(memoryString)
	if err != nil {
		return resource.Quantity{}, fmt.Errorf("failed to parse memory quantity: %w", err)
	}
```

Reply:
```go
quantity := resource.NewQuantity(memoryGiB, resource.BinarySI)
```
would save the format/parse roundtrip + error checking

Reply:
but adds converting float32 to int64

Reply:
Sure, but right now we do `float32 -> string -> int64`, where [int64 conversion lives in `ParseQuantity`](https://github.com/kubernetes/apimachinery/blob/v0.31.2/pkg/api/resource/quantity.go#L277). I agree that the `memoryGiB` would need scaling, so `quantity := resource.NewQuantity(int64(memoryGiB)*1024*1024*1024, resource.BinarySI)`. Alternativly, we care about the precision there is also `ParseDecimalQuantity`.

Reply:
you are looking at an outdated code :)